### ISSUE TITLE
Internal Errors en algunas paginas de detalle

### ISSUE THREAD

**Issue by mgaitan** [mgaitan/preciosa#70]
Ejemplo 
http://preciosdeargentina.com.ar/905-te/11151-te-50-saquitos-50-un
http://preciosdeargentina.com.ar/849-arroz/10135-arroz-gallo-doble-carolina-paq-500-grm

**Comment by mgaitan** [comment]
``` python
[14/Feb/2014 06:13:34] "GET /866-pegamentos/10358-topetina-almohadilla-adhesiva-cirucular-color-blanca-4-un HTTP/1.1" 500 59
Traceback (most recent call last):
  File "/usr/lib/python2.7/wsgiref/handlers.py", line 85, in run
    self.result = application(self.environ, self.start_response)
  File "/virtualenvs/preciosa/local/lib/python2.7/site-packages/django/core/handlers/wsgi.py", line 206, in __call__
    response = self.get_response(request)
  File "/virtualenvs/preciosa/local/lib/python2.7/site-packages/django/core/handlers/base.py", line 155, in get_response
    response = self.handle_uncaught_exception(request, resolver, sys.exc_info())
  File "/virtualenvs/preciosa/local/lib/python2.7/site-packages/django/core/handlers/base.py", line 238, in handle_uncaught_exception
    return callback(request, **param_dict)
  File "/virtualenvs/preciosa/local/lib/python2.7/site-packages/django/utils/decorators.py", line 99, in _wrapped_view
    response = view_func(request, *args, **kwargs)
  File "/virtualenvs/preciosa/local/lib/python2.7/site-packages/django/views/defaults.py", line 46, in server_error
    return http.HttpResponseServerError(template.render(Context({})))
  File "/virtualenvs/preciosa/local/lib/python2.7/site-packages/django/template/base.py", line 140, in render
    return self._render(context)
  File "/virtualenvs/preciosa/local/lib/python2.7/site-packages/django/template/base.py", line 134, in _render
    return self.nodelist.render(context)
  File "/virtualenvs/preciosa/local/lib/python2.7/site-packages/django/template/base.py", line 840, in render
    bit = self.render_node(node, context)
  File "/virtualenvs/preciosa/local/lib/python2.7/site-packages/django/template/debug.py", line 78, in render_node
    return node.render(context)
  File "/virtualenvs/preciosa/local/lib/python2.7/site-packages/django/template/loader_tags.py", line 123, in render
    return compiled_parent._render(context)
  File "/virtualenvs/preciosa/local/lib/python2.7/site-packages/django/template/base.py", line 134, in _render
    return self.nodelist.render(context)
  File "/virtualenvs/preciosa/local/lib/python2.7/site-packages/django/template/base.py", line 840, in render
    bit = self.render_node(node, context)
  File "/virtualenvs/preciosa/local/lib/python2.7/site-packages/django/template/debug.py", line 78, in render_node
    return node.render(context)
  File "/virtualenvs/preciosa/local/lib/python2.7/site-packages/django/template/loader_tags.py", line 123, in render
    return compiled_parent._render(context)
  File "/virtualenvs/preciosa/local/lib/python2.7/site-packages/django/template/base.py", line 134, in _render
    return self.nodelist.render(context)
  File "/virtualenvs/preciosa/local/lib/python2.7/site-packages/django/template/base.py", line 840, in render
    bit = self.render_node(node, context)
  File "/virtualenvs/preciosa/local/lib/python2.7/site-packages/django/template/debug.py", line 78, in render_node
    return node.render(context)
  File "/virtualenvs/preciosa/local/lib/python2.7/site-packages/django/template/loader_tags.py", line 123, in render
    return compiled_parent._render(context)
  File "/virtualenvs/preciosa/local/lib/python2.7/site-packages/django/template/base.py", line 134, in _render
    return self.nodelist.render(context)
  File "/virtualenvs/preciosa/local/lib/python2.7/site-packages/django/template/base.py", line 840, in render
    bit = self.render_node(node, context)
  File "/virtualenvs/preciosa/local/lib/python2.7/site-packages/django/template/debug.py", line 78, in render_node
    return node.render(context)
  File "/virtualenvs/preciosa/local/lib/python2.7/site-packages/django/template/loader_tags.py", line 62, in render
    result = block.nodelist.render(context)
  File "/virtualenvs/preciosa/local/lib/python2.7/site-packages/django/template/base.py", line 840, in render
    bit = self.render_node(node, context)
  File "/virtualenvs/preciosa/local/lib/python2.7/site-packages/django/template/debug.py", line 78, in render_node
    return node.render(context)
  File "/virtualenvs/preciosa/local/lib/python2.7/site-packages/django/template/loader_tags.py", line 62, in render
    result = block.nodelist.render(context)
  File "/virtualenvs/preciosa/local/lib/python2.7/site-packages/django/template/base.py", line 840, in render
    bit = self.render_node(node, context)
  File "/virtualenvs/preciosa/local/lib/python2.7/site-packages/django/template/debug.py", line 78, in render_node
    return node.render(context)
  File "/virtualenvs/preciosa/local/lib/python2.7/site-packages/django/template/loader_tags.py", line 62, in render
    result = block.nodelist.render(context)
  File "/virtualenvs/preciosa/local/lib/python2.7/site-packages/django/template/base.py", line 840, in render
    bit = self.render_node(node, context)
  File "/virtualenvs/preciosa/local/lib/python2.7/site-packages/django/template/debug.py", line 78, in render_node
    return node.render(context)
  File "/virtualenvs/preciosa/local/lib/python2.7/site-packages/django/template/defaulttags.py", line 447, in render
    six.reraise(*exc_info)
  File "/virtualenvs/preciosa/local/lib/python2.7/site-packages/django/template/defaulttags.py", line 433, in render
    url = reverse(view_name, args=args, kwargs=kwargs, current_app=context.current_app)
  File "/virtualenvs/preciosa/local/lib/python2.7/site-packages/django/core/urlresolvers.py", line 509, in reverse
    return iri_to_uri(resolver._reverse_with_prefix(view, prefix, *args, **kwargs))
  File "/virtualenvs/preciosa/local/lib/python2.7/site-packages/django/core/urlresolvers.py", line 429, in _reverse_with_prefix
    (lookup_view_s, args, kwargs, len(patterns), patterns))
NoReverseMatch: Reverse for 'home' with arguments '()' and keyword arguments '{}' not found. 0 pattern(s) tried: []
Traceback (most recent call last):
  File "/usr/lib/python2.7/wsgiref/handlers.py", line 85, in run
    self.result = application(self.environ, self.start_response)
  File "/virtualenvs/preciosa/local/lib/python2.7/site-packages/django/core/handlers/wsgi.py", line 206, in __call__
    response = self.get_response(request)
  File "/virtualenvs/preciosa/local/lib/python2.7/site-packages/django/core/handlers/base.py", line 155, in get_response
    response = self.handle_uncaught_exception(request, resolver, sys.exc_info())
  File "/virtualenvs/preciosa/local/lib/python2.7/site-packages/django/core/handlers/base.py", line 238, in handle_uncaught_exception
    return callback(request, **param_dict)
  File "/virtualenvs/preciosa/local/lib/python2.7/site-packages/django/utils/decorators.py", line 99, in _wrapped_view
    response = view_func(request, *args, **kwargs)
  File "/virtualenvs/preciosa/local/lib/python2.7/site-packages/django/views/defaults.py", line 46, in server_error
    return http.HttpResponseServerError(template.render(Context({})))
  File "/virtualenvs/preciosa/local/lib/python2.7/site-packages/django/template/base.py", line 140, in render
    return self._render(context)
  File "/virtualenvs/preciosa/local/lib/python2.7/site-packages/django/template/base.py", line 134, in _render
    return self.nodelist.render(context)
  File "/virtualenvs/preciosa/local/lib/python2.7/site-packages/django/template/base.py", line 840, in render
    bit = self.render_node(node, context)
  File "/virtualenvs/preciosa/local/lib/python2.7/site-packages/django/template/debug.py", line 78, in render_node
    return node.render(context)
  File "/virtualenvs/preciosa/local/lib/python2.7/site-packages/django/template/loader_tags.py", line 123, in render
    return compiled_parent._render(context)
  File "/virtualenvs/preciosa/local/lib/python2.7/site-packages/django/template/base.py", line 134, in _render
    return self.nodelist.render(context)
  File "/virtualenvs/preciosa/local/lib/python2.7/site-packages/django/template/base.py", line 840, in render
    bit = self.render_node(node, context)
  File "/virtualenvs/preciosa/local/lib/python2.7/site-packages/django/template/debug.py", line 78, in render_node
    return node.render(context)
  File "/virtualenvs/preciosa/local/lib/python2.7/site-packages/django/template/loader_tags.py", line 123, in render
    return compiled_parent._render(context)
  File "/virtualenvs/preciosa/local/lib/python2.7/site-packages/django/template/base.py", line 134, in _render
    return self.nodelist.render(context)
  File "/virtualenvs/preciosa/local/lib/python2.7/site-packages/django/template/base.py", line 840, in render
    bit = self.render_node(node, context)
  File "/virtualenvs/preciosa/local/lib/python2.7/site-packages/django/template/debug.py", line 78, in render_node
    return node.render(context)
  File "/virtualenvs/preciosa/local/lib/python2.7/site-packages/django/template/loader_tags.py", line 123, in render
    return compiled_parent._render(context)
  File "/virtualenvs/preciosa/local/lib/python2.7/site-packages/django/template/base.py", line 134, in _render
    return self.nodelist.render(context)
  File "/virtualenvs/preciosa/local/lib/python2.7/site-packages/django/template/base.py", line 840, in render
    bit = self.render_node(node, context)
  File "/virtualenvs/preciosa/local/lib/python2.7/site-packages/django/template/debug.py", line 78, in render_node
    return node.render(context)
  File "/virtualenvs/preciosa/local/lib/python2.7/site-packages/django/template/loader_tags.py", line 62, in render
    result = block.nodelist.render(context)
  File "/virtualenvs/preciosa/local/lib/python2.7/site-packages/django/template/base.py", line 840, in render
    bit = self.render_node(node, context)
  File "/virtualenvs/preciosa/local/lib/python2.7/site-packages/django/template/debug.py", line 78, in render_node
    return node.render(context)
  File "/virtualenvs/preciosa/local/lib/python2.7/site-packages/django/template/loader_tags.py", line 62, in render
    result = block.nodelist.render(context)
  File "/virtualenvs/preciosa/local/lib/python2.7/site-packages/django/template/base.py", line 840, in render
    bit = self.render_node(node, context)
  File "/virtualenvs/preciosa/local/lib/python2.7/site-packages/django/template/debug.py", line 78, in render_node
    return node.render(context)
  File "/virtualenvs/preciosa/local/lib/python2.7/site-packages/django/template/loader_tags.py", line 62, in render
    result = block.nodelist.render(context)
  File "/virtualenvs/preciosa/local/lib/python2.7/site-packages/django/template/base.py", line 840, in render
    bit = self.render_node(node, context)
  File "/virtualenvs/preciosa/local/lib/python2.7/site-packages/django/template/debug.py", line 78, in render_node
    return node.render(context)
  File "/virtualenvs/preciosa/local/lib/python2.7/site-packages/django/template/defaulttags.py", line 447, in render
    six.reraise(*exc_info)
  File "/virtualenvs/preciosa/local/lib/python2.7/site-packages/django/template/defaulttags.py", line 433, in render
    url = reverse(view_name, args=args, kwargs=kwargs, current_app=context.current_app)
  File "/virtualenvs/preciosa/local/lib/python2.7/site-packages/django/core/urlresolvers.py", line 509, in reverse
    return iri_to_uri(resolver._reverse_with_prefix(view, prefix, *args, **kwargs))
  File "/virtualenvs/preciosa/local/lib/python2.7/site-packages/django/core/urlresolvers.py", line 429, in _reverse_with_prefix
    (lookup_view_s, args, kwargs, len(patterns), patterns))
NoReverseMatch: Reverse for 'home' with arguments '()' and keyword arguments '{}' not found. 0 pattern(s) tried: []
```

esto me da `deploy.py run debug`  (que sirve el el runserver de django directamente, bypasseando nginx)

**Comment by mgaitan** [comment]
Parece que se arregló en parte Ahora parece ser otro problema, ya no es la ausencia de url llamada `home` sino otra cosa. No se ve "Internal server error" sino "a server error has occurred." capturado por django. 

http://preciosdeargentina.com.ar:8000/846-bizcochos/10112-palmeras-bizcochos-dulces-de-hojaldre-150-grs

no se ve ningun traceback (?)

**Comment by ferduz** [comment]
Martin, Like I told you, no tengo ni la min idea de python.... lo sumo puedo acercarte algo...

---

 Currently, the default server_error view in django.views.defaults renders using the following code:

return http.HttpResponseServerError(t.render(Context({})))

However, this prevents the context processors specified in SETTINGS.TEMPLATE_CONTEXT_PROCESSORS from being loaded. For example, if a base.html might include branding that includes the current date and time (which is passed in through a custom context processor). Currently, if we try to extend base.html in the 500.html template, we get an error because the context variable for the date and time isn't loaded.

The best solution seems to be similar to the way the page_not_found view renders:

return http.HttpResponseServerError(t.render(RequestContext(request)))

Although the page_not_found view also passes in a request_path variable to RequestContext, I'm not sure if this is necessary.

source: https://code.djangoproject.com/ticket/5617

**Comment by mgaitan** [comment]
fixed in bc0a188
